### PR TITLE
chore: add pull request checklist and commit message documentation

### LINKED ISSUE #1168: create a pull request template
date the PR template to include a checklist of definition of done, make a list of those things that need to be there, codestyle, best practices, unit test, e2e test, etc

https://docs.github.com/en/communities/using-templates-to-encourage-useful-issues-and-pull-requests/creating-a-pull-request-template-for-your-repository

created separate PRs to track blocked work:
- relates to: https://github.com/hashgraph/solo/issues/1675
- relates to: https://github.com/hashgraph/solo/issues/1676

### PR DESCRIPTION
**Pull request by jeromy-cannon**
## Description

This pull request changes the following:

* add pull request checklist and commit message documentation

### Related Issues

* Closes #1168

### CODE REVIEW

**Review by JeffreyDallas — APPROVED** [review]

File: .github/pull_request_template.md [hunk 1]
```diff
@@ -4,6 +4,58 @@ This pull request changes the following:
...
+2. The title is prefixed with one of the following:
+
+| Prefix    | Description                                         | Semantic Version Update | Captured in Release Notes |
+|-----------|-----------------------------------------------------|-------------------------|---------------------------|
+| feat:     | a new feature                                       | MINOR                   | Yes                       |
+| fix:      | a bug fix                                           | PATCH                   | Yes                       |
+| perf:     | performance                                         | PATCH                   | Yes                       |
+| refactor: | code change that isn't feature or fix               | none                    | No                        |
+| test:     | adding missing tests                                | none                    | No                        |
+| docs:     | changes to documentation                            | none                    | No                        |
```
Comment (nathanklick) [inline .github/pull_request_template.md:55]:
```suggestion
| docs:     | changes to documentation                            | none                    | Yes                        |
```

**Review by nathanklick — APPROVED** [review]
